### PR TITLE
Populate songs without local manifest

### PR DESCRIPTION
**Pull request by epeach**
[Brief spec here.](https://docs.google.com/document/d/16xiyt3gw7XhEcN3hoadejib2iiZwvYTzns0Y2-yCbqk/edit)
*Removes reliance on hard-coded manifest
*Fetches songs to display and load in song-selector
*Removes song register from onP5Preload and moves to song selector.

NOTE: This does not have an elegant fail case for when the song manifest, song metadata, and available mp3 are out of sync. Current behavior is just not playing song for missing mp3 file or characters not dancing for missing metadata.

### CODE REVIEW

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 1]
```diff
@@ -21,67 +20,60 @@ const styles = {
...
     if (this.props.isProjectLevel) {
       //Save song to project
       project.saveSelectedSong(song);
     }
-  };
+  }
+
+  componentDidMount() {
+    fetch(`/api/v1/sound-library/hoc_song_meta/songManifest.json`)
+      .then((response) => {return response.json();})
```
Comment (islemaster) [inline apps/src/dance/DanceVisualizationColumn.jsx]:
```suggestion
      .then(response => response.json())
```

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 2]
```diff
@@ -21,67 +20,60 @@ const styles = {
...
+      .then((data) => {this.parseSongOptions(data);});
```
Comment (islemaster) [inline apps/src/dance/DanceVisualizationColumn.jsx]:
```suggestion
      .then(data => this.parseSongOptions(data));
```

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 3]
```diff
@@ -21,67 +20,60 @@ const styles = {
...
+  parseSongOptions(data) {
+    let songs = {};
+    if (data) {
+      data.songs.forEach((song) => {
+          songs[song.id] = {title: song.text, url: song.url};
+      });
+    }
+    this.setState({songsData: songs});
+
+    this.loadSong(this.props.selectedSong);
```
Comment (islemaster) [inline apps/src/dance/DanceVisualizationColumn.jsx]: `setState` is not guaranteed to be synchronous; if `loadSong` depends on it, it should happen in a callback:
```js
this.setState({songsData: songs}, () => {
  this.loadSong(this.props.selectedSong);
});
```

File: apps/src/gamelab/GameLabVisualizationColumn.jsx [hunk 4]
```diff
@@ -56,13 +59,29 @@ const SongSelector = Radium(class extends React.Component {
...
+      .then((response) => {return response.json();})
+      .then((data) => {this.parseSongOptions(data);});
+  }
+
+  parseSongOptions(data) {
+    let songs = {};
+    if (data) {
+      data.songs.forEach((song) => {
+        songs[song.id] = {title: song.text, url: song.url};
+      });
```
Comment (islemaster) [inline apps/src/gamelab/GameLabVisualizationColumn.jsx:73]: It looks like there's some repeated fetch/parsing logic here - it'd be nice to deduplicate this.

Comment (epeach) [reply apps/src/gamelab/GameLabVisualizationColumn.jsx:73]: It looks like the old-runtime is going to be deprecated and removed, so this whole file will be removed soon.

File: apps/src/dance/DanceVisualizationColumn.jsx [hunk 5]
```diff
@@ -21,67 +20,60 @@ const styles = {
...
 
   render() {
     return (
       <div>
         <label><b>{i18n.selectSong()}</b></label>
         <select id="song_selector" style={styles.selectStyle} onChange={this.changeSong} value={this.props.selectedSong}>
-          {Object.keys(songLibrary).map((option, i) => (
-            <option key={i} value={option}>{songLibrary[option].title}</option>
+          {Object.keys(this.state.songsData).map((option, i) => (
+            <option key={i} value={option}>{this.state.songsData[option].title}</option>
```
Comment (joshlory) [inline apps/src/dance/DanceVisualizationColumn.jsx:76]: What is the loading state until the manifest loads?  (Worth extracting `SongSelector.jsx` and adding a story?)

Comment (epeach) [reply apps/src/dance/DanceVisualizationColumn.jsx:76]: At the moment, the dropdown is just empty until the manifest loads. Ship blocking?

Comment (joshlory) [reply apps/src/dance/DanceVisualizationColumn.jsx:76]: Nope!

**Pull request merged**